cli/provider/cmd.go

```diff
@@ -456,15 +475,118 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 	c.logger.Debug("config has been initialised", zap.Any("for cmd", cmd.Name()), zap.Any("config", c.cfg))
 
 	switch cmd.Name() {
+	case "generate", "download":
+		path, err := cmd.Flags().GetString("path")
+		if err != nil {
+			errMsg := "failed to get the path"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+		if path == "" {
+			c.cfg.Contract.Path = path
+		} else {
+			c.cfg.Contract.Path = "./"
+		}
+
+		services, err := cmd.Flags().GetStringSlice("services")
+		if err != nil {
+			errMsg := "failed to get the services"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+		config.SetSelectedServices(c.cfg, services)
+
+		selectedTests, err := cmd.Flags().GetStringSlice("tests")
+		if err != nil {
+			errMsg := "failed to get the tests"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+		config.SetSelectedContractTests(c.cfg, selectedTests)
+
+		if cmd.Name() == "download" {
+			c.cfg.Contract.Driven, err = cmd.Flags().GetString("driven")
+			if err != nil {
+				errMsg := "failed to get the driven flag"
+				utils.LogError(c.logger, err, errMsg)
+				return errors.New(errMsg)
+			}
+			if c.cfg.Contract.Driven == "" {
+				c.cfg.Contract.Driven = "client"
+			}
+
+		}
+		absPath, err := utils.GetAbsPath("./")
+		if err != nil {
+			utils.LogError(c.logger, err, "error while getting absolute path")
+			return errors.New("failed to get the absolute path")
+		}
+		c.cfg.Path = absPath + "/keploy"
+	case "config":
+		path, err := cmd.Flags().GetString("path")
+		if err != nil {
+			errMsg := "failed to get the path"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+		c.cfg.Path = path
 	case "record", "test", "rerecord":
 
+		if cmd.Parent() != nil && cmd.Parent().Name() == "contract" {
+			path, err := cmd.Flags().GetString("path")
+			if err != nil {
+				errMsg := "failed to get the path"
+				utils.LogError(c.logger, err, errMsg)
+				return errors.New(errMsg)
+			}
+			if path == "" {
```

Comment:
Again, it is not required, just call the ToAbs Path if abs path + keploy is required or use GetAbsPath.